Source/WebCore/css/CSSGridTemplateAreasValue.cpp

```diff
@@ -32,49 +32,43 @@
 #include "config.h"
 #include "CSSGridTemplateAreasValue.h"
 
-#include "GridArea.h"
 #include <wtf/FixedVector.h>
 #include <wtf/HashSet.h>
 #include <wtf/text/StringBuilder.h>
 #include <wtf/text/StringHash.h>
 
 namespace WebCore {
 
-CSSGridTemplateAreasValue::CSSGridTemplateAreasValue(NamedGridAreaMap map, size_t rowCount, size_t columnCount)
+CSSGridTemplateAreasValue::CSSGridTemplateAreasValue(CSS::GridTemplateAreas&& areas)
     : CSSValue(ClassType::GridTemplateAreas)
-    , m_map(WTFMove(map))
-    , m_rowCount(rowCount)
-    , m_columnCount(columnCount)
+    , m_areas(WTFMove(areas))
 {
-    ASSERT(m_rowCount);
-    ASSERT(m_columnCount);
+    ASSERT(m_areas.map.rowCount);
+    ASSERT(m_areas.map.columnCount);
 }
 
-Ref<CSSGridTemplateAreasValue> CSSGridTemplateAreasValue::create(NamedGridAreaMap map, size_t rowCount, size_t columnCount)
+CSSGridTemplateAreasValue::CSSGridTemplateAreasValue(const CSS::GridTemplateAreas& areas)
+    : CSSValue(ClassType::GridTemplateAreas)
```

Comment:
Would this be efficient enough?
```suggestion
CSSGridTemplateAreasValue::CSSGridTemplateAreasValue(const CSS::GridTemplateAreas& areas)
    : CSSGridTemplateAreasValue { CSS::GridTemplateAreas { areas } }
{
}
```
I like the idea of not needing to write the assertions twice.